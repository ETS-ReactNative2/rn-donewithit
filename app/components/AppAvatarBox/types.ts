import { ImageSourcePropType, StyleProp, ViewStyle } from 'react-native'

export type AppAvatarBoxProps = {
  title: string
  image: ImageSourcePropType
  subTitle?: string
  style?: StyleProp<ViewStyle>
}
