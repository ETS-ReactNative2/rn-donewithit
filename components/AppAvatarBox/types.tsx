import { ImageSourcePropType } from 'react-native'

export type AppAvatarBoxProps = {
  title: string
  image: ImageSourcePropType
  subTitle?: string
}
