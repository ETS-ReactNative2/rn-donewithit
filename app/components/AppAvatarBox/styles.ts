import { StyleSheet } from 'react-native'
import CONSTS from './consts'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL
  },
  avatar: {
    width: CONSTS.AVATAR_SIZE,
    height: CONSTS.AVATAR_SIZE,
    borderRadius: CONSTS.AVATAR_SIZE / 2,
    marginRight: CONSTS.PADDING_HORIZONTAL / 2
  },
  infoBox: {
    flexShrink: 1
  },
  text: {
    fontSize: CONSTS.FONT_SIZE
  },
  subTitle: {
    color: COLORS.GRAY,
    marginTop: CONSTS.SUBTITLE_MARGIN_TOP
  }
})

export default styles
