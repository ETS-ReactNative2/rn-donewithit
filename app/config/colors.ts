const COLORS = {
  PRIMARY: 'tomato',
  SECONDARY: 'turquoise',
  BACKGROUND_COLOR: '#f8f4f4',
  INPUT_BACKGROUND_COLOR: '#f9f2f2',
  DANGER: '#ff5252',
  GRAY: '#6e6969',
  LIGHT_GRAY: '#f8f4f4',
  WHITE: '#fff',
  BLACK: '#000',
  SILVER: '#C0C0C0',
  YELLOW: '#ffe66d'
}

export default COLORS
