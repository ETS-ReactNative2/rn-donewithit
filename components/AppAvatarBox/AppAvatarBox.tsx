import React from 'react'
import { View, Image, StyleSheet } from 'react-native'
import AppText from '../AppText'
import { AppAvatarBoxProps } from './types'
import CONSTS from './consts'
import COLORS from '../../config/colors'

function AppAvatarBox({ title, image, subTitle, style }: AppAvatarBoxProps) {
  return (
    <View style={[style, styles.container]}>
      <Image style={styles.avatar} source={image} />
      <View style={styles.infoBox}>
        <AppText style={styles.text}>{title}</AppText>
        <AppText style={[styles.text, styles.subTitle]}>{subTitle}</AppText>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL
  },
  avatar: {
    width: CONSTS.AVATAR_SIZE,
    height: CONSTS.AVATAR_SIZE,
    borderRadius: CONSTS.AVATAR_SIZE / 2,
    marginRight: CONSTS.PADDING_HORIZONTAL / 2
  },
  infoBox: {
    flexShrink: 1
  },
  text: {
    fontSize: CONSTS.FONT_SIZE
  },
  subTitle: {
    color: COLORS.GRAY,
    marginTop: CONSTS.SUBTITLE_MARGIN_TOP
  }
})

export default AppAvatarBox
