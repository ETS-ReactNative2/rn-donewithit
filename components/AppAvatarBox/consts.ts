const CONSTS = {
  FONT_SIZE: 18,
  PADDING_HORIZONTAL: 20,
  AVATAR_SIZE: 80,
  SUBTITLE_MARGIN_TOP: 5
}

export default CONSTS
