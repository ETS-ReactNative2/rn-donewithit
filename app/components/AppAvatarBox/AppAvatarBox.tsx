import React from 'react'
import { View, Image } from 'react-native'
import AppText from '../AppText'
import { AppAvatarBoxProps } from './types'
import styles from './styles'

function AppAvatarBox({ title, image, subTitle, style }: AppAvatarBoxProps) {
  return (
    <View style={[style, styles.container]}>
      <Image style={styles.avatar} source={image} />
      <View style={styles.infoBox}>
        <AppText style={styles.text}>{title}</AppText>
        <AppText style={[styles.text, styles.subTitle]}>{subTitle}</AppText>
      </View>
    </View>
  )
}

export default AppAvatarBox
