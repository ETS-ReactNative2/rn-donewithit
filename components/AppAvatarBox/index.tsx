import AppAvatarBox from './AppAvatarBox'
export default AppAvatarBox
