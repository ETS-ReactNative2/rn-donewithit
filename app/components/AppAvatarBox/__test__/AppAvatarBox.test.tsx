import React from 'react'
import renderer from 'react-test-renderer'
import { StyleSheet } from 'react-native'
import AppAvatarBox from '../index'
import COLORS from '../../../config/colors'
import { MOSH } from '../../../config/images'

const styles = StyleSheet.create({
  avatarBox: {
    backgroundColor: COLORS.WHITE,
    paddingVertical: 20,
    marginTop: 30,
    marginBottom: 40
  }
})

const props = {
  style: styles.avatarBox,
  title: 'Junwen Huang',
  subTitle: 'jimvinhuang@gmail.com',
  image: MOSH
}

describe('AppAvatarBox', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppAvatarBox {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
