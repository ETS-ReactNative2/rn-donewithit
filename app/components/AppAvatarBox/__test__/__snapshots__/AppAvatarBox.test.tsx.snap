// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAvatarBox renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "marginBottom": 40,
        "marginTop": 30,
        "paddingVertical": 20,
      },
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 40,
        "height": 80,
        "marginRight": 10,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      Junwen Huang
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#6e6969",
              "marginTop": 5,
            },
          ],
        ]
      }
    >
      jimvinhuang@gmail.com
    </Text>
  </View>
</View>
`;
